File: proposed-projects/round-trip-experiment/reverse/20260909T012246Z_fbe56704/20260910T051241Z_fa3c6dd0/ported_implementation/src/rules-builder/parse-space.ts
import { charAt } from './char-at.ts';

export const parseSpace = (src: string, pos: number, newlineOk: boolean): number => {
  while (
    [' ', '\t', '#'].includes(charAt(src, pos)) ||
    (newlineOk && ['\r', '\n'].includes(charAt(src, pos)))
  ) {
    if (charAt(src, pos) === '#') {
      while (charAt(src, pos) && !['\r', '\n'].includes(charAt(src, pos))) {
        pos += 1;
      }
    } else {
      pos += 1;
    }
  }
  return pos;
};

File: packages/gbnf-experiment/data/20260909T133702Z_8305aa82/ported_implementation/gbnf/rules_builder/parse_space.py
def parse_space(src: str, pos: int, newline_ok: bool) -> int:
    def at(i: int) -> str:
        return src[i] if 0 <= i < len(src) else ""

    while at(pos) in (" ", "\t", "#") or (newline_ok and at(pos) in ("\r", "\n")):
        if at(pos) == "#":
            while at(pos) and at(pos) != "\r" and at(pos) != "\n":
                pos += 1
        else:
            pos += 1
    return pos
